backend/src/main/java/c/team/session/administration/controller/SessionController.java

```diff
@@ -90,13 +92,20 @@ public Message guestApprovalResponse(@DestinationVariable String sessionApproval
 
     @MessageMapping("/socket/session/{sessionId}/quiz")    // Here leader sends a quiz
     @SendTo("/topic/session/{sessionId}/quiz")      // Here participants subscribe to receive quiz
-    public Message sendQuizToParticipants(@DestinationVariable String sessionId, @Payload final Message message){
-        if(message.getType() != MessageType.QUIZ)
+    public Message sendQuizToParticipants(@DestinationVariable String sessionId, @Payload final Message message) {
+        if (message.getType() != MessageType.QUIZ)
             throw new InvalidMessageTypeException();
         sessionsService.addMessageToSessionLog(sessionId, message);
         return message;
     }
 
+    @MessageMapping("/socket/session/{sessionId}/reaction")
+    @SendTo("/topic/session/{sessionId}/reaction")
+    public String sendReaction(@DestinationVariable String sessionId, @Payload String reactionString) {
+        reactionService.saveReaction(sessionId, reactionString);
+        return reactionString;
+    }
+
```

Comment:
Tutaj bym tylko jeszcze dorzucił loggowanie tego do timeline'u - może jako wiadomość nowego typu EMOJI, o minimalnej zawartości (bo timeline logguje tylko wiadomości chyba). Tylko opakować to w Message.